[12.0] auth_saml: Refactoring using pysaml2

eLBati opened the pull request.

theangryangel committed:
Refactoring using pysaml2

eLBati committed:
FIX auth_saml when reading/writing config_parameter: False values are not stored
Also, the default value should be False anyway, for security

eLBati committed:
FIX setupt external_dependencies_override
IMP README

eLBati committed:
Handle ir_attachment.location != file
FIX TypeError: ord() expected string of length 1, but int found
FIX AttributeError: 'res.users' object has no attribute 'password_crypt'
FIX as we are not correctly generating metadata, don't require signed responses, since default metadata configurations do not require them
FIX RecursionError: maximum recursion depth exceeded

eLBati commented:
@theangryangel I'm adding fixes which should be also ported to v13

theangryangel commented:
Noted. I’m away for a week, I’ll make sure it’s one of the first things I take a look at when im back :+1:

theangryangel commented on auth_saml/models/auth_saml.py:
```diff
@@ -1,78 +1,331 @@
+                    # TODO: as we are not correctly generating metadata, don't require
+                    #  signed responses, since default metadata configurations
+                    #  do not require them
+                    "want_response_signed": False,
```
Hi @eLBati can you give a bit more detail on this change? It's something I have working in live, if there's a bug with a particular IDP I can see if I can help out? (for ref I've got a copy working in 12.0 against Duo)

theangryangel commented:
@eLBati FYI I've made some changes to our 13.0 port based on our internal day-to-day usage, it includes a couple of changes;

* Additional (optional) module to filter saml auth providers by website (solves a problem we have where it turns out Duo just ignores the ACS URL it seems, and we need multiple SAML providers configured)
* Fixes the signature issue you pointed out and makes the signature algorithm configurable
* Fixes a couple of issues with with non-fixed base urls
* Allows a user to have multiple SAML providers 

I'll update the 12.0 copy at https://github.com/GlodoUK/server-auth/tree/feat/12.0-mig-auth_saml-pysaml once we've had it running for a few days. I'm not sure if you'll want to update your copy from that

eLBati commented:
@theangryangel thanks, let me know when you have uploaded the changes

ovnicraft commented:
@eLBati 
would be nice include controller to response /metadata, what do you think ?

Nice ! https://github.com/OCA/server-auth/pull/234/files#diff-d051ee3f0ce073219b1c7ebc99bfa35206c9ebeefbf62df714fb8c800ad0310cR195

moylop260 committed:
[REF] auth_saml: Fix flake8 checks
[REF] requirements.txt: Use new requirement package for auth_saml
[REF] .travis.yml: Using py3.6 compatible with pysaml2

moylop260 commented:
@theangryangel @eLBati 

I just fixed the flake8 checks raised in travis in last commit.

moylop260 commented:
@theangryangel @eLBati
I have changed the requirements.txt in order to install the module correctly in the CI
Pinning the version in order to avoid future errors with incompatible releases

moylop260 commented on auth_saml/models/saml_request.py:
```diff
@@ -0,0 +1,16 @@
+
+
+class SamlToken(models.TransientModel):
+    _name = "auth_saml.request"
```
If the provider model is called `auth.saml.provider`

What about calling this one:

```suggestion
    _name = "auth.saml.request"
```

moylop260 commented on auth_saml/__manifest__.py:
```diff
@@ -13,7 +14,7 @@
+        "python": ["saml2"],
     },
     "demo": [
         'demo/auth_saml.xml',
```
demo file  show the following warning:

[2020-12-18 04:53:45,018 12075 WARNING auth odoo.models: auth.saml.provider.create() with unknown fields: sp_metadata, sp_pkey](https://travis-ci.com/github/OCA/server-auth/jobs/462174698#L1348)

yvaucher commented on auth_saml/__manifest__.py:
```diff
@@ -13,7 +14,7 @@
         'web',
     ],
     "external_dependencies": {
-        "python": ["lasso"],
```
Dependency is mentionned in the readme, you will want to change it there too.

yvaucher replied:
my bad it's in INSTALL.rst, it will be updated automatically

yvaucher commented on auth_saml/models/auth_saml.py:
```diff
@@ -1,78 +1,331 @@
+
+    sp_pem = fields.Binary(
+        string="Odoo Certificate",
+        help="Combined certificate and private key used by Odoo to sign",
```
Can you state "pem" in the help text to help explaining what kind of file is expected?

yvaucher reviewed (approved):
Tested locally with a KeyCloak config.

Worked for me to login a user.

luisg123v commented:
Hi @eLBati,

Is there any news regarding this PR? Do you plan to continue it?

theangryangel commented:
I can open a refresh PR and incorporate the changes to both the 12.0 and the original 13.0 branch this came from, if there’s interest. I won’t get around to it got a few days tho

eLBati commented:
Actually we abandoned the project, so I can close this if you want to move to another PR.
Thanks

luisg123v commented:
@eLBati  yes, please. If not working on this anymore, please close it so the community may now it won't be finished.

yvaucher commented:
We should definitely use pysaml instead of lasso, it eases the installation of this module.

@eLBati what was the missing part on this that makes you think it's not ready?

eLBati commented:
@yvaucher just the open remarks of this PR.
We successfully used the module with Azure AD, then the customer moved to other shores

ovnicraft commented:
@yvaucher module is usable, so i am in favor to open it again an merge it
